### PR TITLE
Assistant Builder for DustAppRun Configuration

### PR DESCRIPTION
**Pull request by spolu**
Adds support for DustAppRun to AssistantBuilder

Deploynent:
- init_db on front
- deploy front

Screenshot from 2023-10-10 15-52-28
Screenshot from 2023-10-10 15-52-32
Screenshot from 2023-10-10 15-52-38

### CODE REVIEW

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 1]
```diff
@@ -879,6 +968,74 @@ export default function AssistantBuilder({
...
+                <div className="flex flex-row items-center space-x-2 pt-6">
+                  <div className="text-sm font-semibold text-element-900">
+                    Dust App:
+                  </div>
+                  <DropdownMenu>
+                    <DropdownMenu.Button>
+                      <Button
+                        type="select"
+                        labelVisible={true}
+                        disabled={builderState.dataSourceMode !== "GENERIC"}
```
Comment (fontanierh) [inline front/components/assistant_builder/AssistantBuilder.tsx:997]: Probably not for now, but it's quite confusing to have the DS become disabled / the dust app become disabled without any explanation

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 2]
```diff
@@ -90,6 +95,21 @@ export type AssistantBuilderDataSourceConfiguration = {
   isSelectAll: boolean;
 };
 
+// DustAppRun Action
+
+const DUST_APP_MODES = ["GENERIC", "SELECTED"] as const;
+type DustAppMode = (typeof DUST_APP_MODES)[number];
+const DUST_APP_MODE_TO_LABEL: Record<DataSourceMode, string> = {
+  GENERIC: "None (Generic model)",
```
Comment (fontanierh) [inline front/components/assistant_builder/AssistantBuilder.tsx:103]: I think `(Generic model)` is strange. Just `None` is probably fine ?

File: front/lib/models/assistant/agent.ts [hunk 3]
```diff
@@ -185,6 +192,16 @@ AgentConfiguration.belongsTo(AgentRetrievalConfiguration, {
   foreignKey: { name: "retrievalConfigurationId", allowNull: true }, // null = no retrieval action set for this Agent
 });
 
+// Agent config <> DustAppRun config
```
Comment (fontanierh) [inline front/lib/models/assistant/agent.ts:195]: Do we want to enforce that Dust app is mutually exclusive with retrieval config ?

File: front/pages/api/w/[wId]/assistant/agent_configurations/index.ts [hunk 4]
```diff
@@ -214,6 +219,13 @@ export async function createOrUpgradeAgentConfiguration(
       dataSources: action.dataSources,
     });
   }
+  if (action && action.type === "dust_app_run_configuration") {
```
Comment (fontanierh) [inline front/pages/api/w/[wId]/assistant/agent_configurations/index.ts:222]: I think we can do if/else if/else never here ?

File: front/pages/w/[wId]/builder/assistants/[aId]/index.tsx [hunk 5]
```diff
@@ -157,17 +183,28 @@ export default function EditAssistant({
     }
   }
 
+  let dustAppMode: AssistantBuilderInitialState["dustAppMode"] = "GENERIC";
+
+  console.log(agentConfiguration.action);
```
Comment (fontanierh) [inline front/pages/w/[wId]/builder/assistants/[aId]/index.tsx:188]: nit

**Review by fontanierh — COMMENTED** [review]
very nice

File: front/components/assistant_builder/AssistantBuilder.tsx (hunk 2 above)
Comment (spolu) [reply front/components/assistant_builder/AssistantBuilder.tsx:103]: Yeah we can use None will update

File: front/components/assistant_builder/AssistantBuilder.tsx (hunk 1 above)
Comment (spolu) [reply front/components/assistant_builder/AssistantBuilder.tsx:997]: cc @Duncid any idea heere?

File: front/lib/models/assistant/agent.ts (hunk 3 above)
Comment (spolu) [reply front/lib/models/assistant/agent.ts:195]: This is mechanically enforced by our code, but we could use a hook indeed

File: front/pages/api/w/[wId]/assistant/agent_configurations/index.ts (hunk 4 above)
Comment (spolu) [reply front/pages/api/w/[wId]/assistant/agent_configurations/index.ts:222]: What's the syntax for the else never ?

File: front/components/assistant_builder/AssistantBuilder.tsx (hunk 1 above)
Comment (Duncid) [reply front/components/assistant_builder/AssistantBuilder.tsx:997]: Don't understand what the problem is 😅

**Pull request merged**